apps/tlon-mobile/src/screens/ChannelSearchScreen.tsx

```diff
@@ -0,0 +1,82 @@
+import type { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { useChannelSearch } from '@tloncorp/shared/dist';
+import * as db from '@tloncorp/shared/dist/db';
+import { XStack, YStack } from '@tloncorp/ui';
+// TODO: update references to dist
+import { Button, SearchBar, SearchResults } from '@tloncorp/ui/src/index';
+import { useCallback, useLayoutEffect, useState } from 'react';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import type { HomeStackParamList } from '../types';
+
+type ChannelSearchProps = NativeStackScreenProps<
+  HomeStackParamList,
+  'ChannelSearch'
+>;
+
+export default function ChannelSearch({
+  route,
+  navigation,
+}: ChannelSearchProps) {
+  const { channel } = route.params;
+  const group = db.useGroupByChannel(channel.id);
+  const [query, setQuery] = useState('');
+  const { posts, loading, errored, hasMore, loadMore, searchedThroughDate } =
+    useChannelSearch(channel.id, query);
+
+  // handle full screen view without bottom nav, resets on dismout
+  useLayoutEffect(() => {
+    navigation.getParent()?.setOptions({
+      headerShown: false,
+      tabBarStyle: { display: 'none' },
+    });
+
+    return () => {
+      navigation.getParent()?.setOptions({
+        tabBarStyle: undefined,
+      });
+    };
+  }, [navigation]);
+
+  const navigateToPost = useCallback(
+    (post: db.PostWithRelations) => {
+      navigation.navigate('Channel', {
+        group: group.result!, // something is wrong here?
```

Comment:
I think the issue is just that `result` can be null, right? This is fine for now, since my next chunk of work will make `group` optional entirely on the channel screen, but maybe add a less cryptic comment :P